lib/src/file_util.rs

```diff
@@ -146,6 +162,51 @@ pub fn persist_content_addressed_temp_file<P: AsRef<Path>>(
     }
 }
 
+/// Symlinks are always available on UNIX, but on Windows they may not be
+/// enabled
+#[cfg(unix)]
+pub fn check_symlink_support() -> Result<bool, std::io::Error> {
+    Ok(true)
+}
+
+/// Symlinks are always available on UNIX, but on Windows they may not be
+/// enabled
+#[cfg(windows)]
+pub fn check_symlink_support() -> Result<bool, std::io::Error> {
+    let hklm = RegKey::predef(HKEY_LOCAL_MACHINE);
+    let sideloading =
+        hklm.open_subkey("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\AppModelUnlock")?;
+    let developer_mode: u32 = sideloading.get_value("AllowDevelopmentWithoutDevLicense")?;
```

Comment:
Will the error message make sense to a user? Maybe we should map it to something more readable (like "Failed to determine if symlinks are supported because <that long key name> wasn't readable"). Actually, none of the callers seem to care about the error, so just return false here instead when we fail to read them instead? How common is it that we would fail to read it?

Reply:
I don't know of a case when reading the registry entry would fail - you'd have to have some weird broken environment. On the other hand, if the environment is that broken I'd want to know about it!

Reply:
> if the environment is that broken I'd want to know about it! I think we should bubble this up like other rare IO-level errors.

Perhaps, tests can be updated to `.unwrap()` the result then? We can also make `TreeState::symlink_support` be evaluated lazily to propagate the error, but I don't know if that'll be worth the added complexity.

Reply:
the tests are using unwrap() :)